src/Ocelot/WebSockets/Middleware/WebSocketsProxyMiddleware.cs

```diff
@@ -117,7 +117,10 @@ private static async Task Proxy(HttpContext context, string serverEndpoint)
                 }
             }
 
-            var destinationUri = new Uri(serverEndpoint);
+            // Only Uris starting with 'ws://' or 'wss://' are supported in System.Net.WebSockets.ClientWebSocket
+            var wsServerEndpoint = serverEndpoint.Replace("https://", "wss://").Replace("http://", "ws://");
+            var destinationUri = new Uri(wsServerEndpoint);
```

Comment:
I would suggest to test the URL string first to detect wrong scheme & protocol:
```suggestion
            if (!serverEndpoint.StartsWith("ws"))
            {
                Logger.LogError("Invalid scheme has detected!");
                serverEndpoint = serverEndpoint
                    .Replace("http://", "ws://")
                    .Replace("https://", "wss://");
            }
            var destinationUri = new Uri(serverEndpoint);
```
We could wrap the `client.ConnectAsync()` by **try-catch-finally** construction to process the exception:
`System.ArgumentException: Only Uris starting with 'ws://' or 'wss://' are supported. (Parameter 'uri')`
But simple **if**-testing is better.

Reply:
This check looks goods, but Logger is not available in `private static async Task Proxy`, and LogError method requires message + exception parameters, maybe LogWarning will suit better?

Reply:
if protocol is invalid, you should raise an exception